library/aes.c

```diff
@@ -331,9 +333,11 @@ static const unsigned char RSb[256] =
     V(71,01,A8,39), V(DE,B3,0C,08), V(9C,E4,B4,D8), V(90,C1,56,64), \
     V(61,84,CB,7B), V(70,B6,32,D5), V(74,5C,6C,48), V(42,57,B8,D0)
 
+#if !defined(MBEDTLS_AES_ONLY_ENCRYPT)
 #define V(a,b,c,d) 0x##a##b##c##d
 static const uint32_t RT0[256] = { RT };
 #undef V
+#endif /* !MBEDTLS_AES_ONLY_ENCRYPT */
```

Comment:
Note: we could also add similar guards to the code paths guarded by `!MBEDTLS_AES_FEWER_TABLES` (just below) and `!MBEDTLS_AES_ROM_TABLES`. This is not essential, as (1) people interested by `ONLY_ENCRYPT` are very likely to enable the other two as well, and (2) anyway this is just about tables that can be garbage-collected by the linker. So I leave it up to you whether you'd like to cover those paths as well, I won't insist if you prefer not to.